drivers/entropy/entropy_mcux_caam.c

```diff
@@ -11,9 +11,17 @@
 #include <zephyr/random/random.h>
 #include <zephyr/init.h>
 #include <zephyr/kernel.h>
+#include <zephyr/cache.h>
 
 #include "fsl_caam.h"
 
+#define ENTROPY_CAAM_CALL(call, ret)	\
+	sys_cache_data_flush_all();	\
+	sys_cache_data_disable();	\
```

Comment:
doesn't the cache disable flush the cache?

Reply:
The cache disable api does not mention that, so I'm not sure what is appropriate

Reply:
We did the inspection of the API and see that it does so it will be safe for us in this instance. 

@carlocaione, the contract on cache_data_disable() doesn't document the flush but we can see that for the Arm architecture it is using the the flush+disable. Should we make that a clear part of the disable API contract that the disable will flush the cache? e.g. update the API documentation.